Dockerfile

```diff
@@ -0,0 +1,41 @@
+FROM centos:centos6 AS builder
+
+# install gcc 6
+RUN yum -y install centos-release-scl && \
+    yum -y install devtoolset-6 devtoolset-6-libatomic-devel
+ENV CC=/opt/rh/devtoolset-6/root/usr/bin/gcc \
+    CPP=/opt/rh/devtoolset-6/root/usr/bin/cpp \
+    CXX=/opt/rh/devtoolset-6/root/usr/bin/g++ \
+    PATH=/opt/rh/devtoolset-6/root/usr/bin:$PATH \
+    LD_LIBRARY_PATH=/opt/rh/devtoolset-6/root/usr/lib64:/opt/rh/devtoolset-6/root/usr/lib:/opt/rh/devtoolset-6/root/usr/lib64/dyninst:/opt/rh/devtoolset-6/root/usr/lib/dyninst:/opt/rh/devtoolset-6/root/usr/lib64:/opt/rh/devtoolset-6/root/usr/lib:$LD_LIBRARY_PATH
+
+# Common development tools and libraries (kitchen sink approach)
+RUN yum groupinstall -y "Development Tools" "Development Libraries"
+
+RUN yum install -y wget rh-mongodb32-boost-devel rh-mongodb32-boost-static
+ENV PATH=/opt/rh/rh-mongodb32/root/usr/bin:$PATH \
+    LD_LIBRARY_PATH=/opt/rh/rh-mongodb32/root/usr/lib64:/opt/rh/rh-mongodb32/root/usr/lib:/opt/rh/rh-mongodb32/root/usr/lib64/dyninst:/opt/rh/rh-mongodb32/root/usr/lib/dyninst:/opt/rh/rh-mongodb32/root/usr/lib64:/opt/rh/rh-mongodb32/root/usr/lib:$LD_LIBRARY_PATH \
+    C_INCLUDE_PATH=/opt/rh/rh-mongodb32/root/usr/include \
+    CPLUS_INCLUDE_PATH=/opt/rh/rh-mongodb32/root/usr/include
+
+
+RUN wget https://cmake.org/files/v3.9/cmake-3.9.0-Linux-x86_64.sh && \
+    chmod +x cmake-3.9.0-Linux-x86_64.sh  && \
+    ./cmake-3.9.0-Linux-x86_64.sh --skip-license --prefix=/usr/local
+
+
+RUN yum install -y zlib-devel libpng-devel epel-release qt5-qtbase-devel
```

Comment:
Is there a  reason there's more than one layer to install modules?

Generally Dockerfile's, when optimizing the size of the image, try to reduce the number of layers.

Reply:
Reducing the size of the builder layer won't make a difference as in the final Docker image is the final image is based on the `runner` layer. 

The reason there are more layers is because the build instructions are copied from https://github.com/The-OpenROAD-Project/alpha-release/blob/master/build/docker/fastroute/Dockerfile and we want to keep the Dockerfile as close as possible to the one that is used currently to be able to test the unified build process.